### PR TITLE
[mono] Implement Vector128.Shuffle using tbl1 on arm64.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -1340,13 +1341,27 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
...
+	if (!strcmp (m_class_get_name (cmethod->klass), "Vector64"))
+		vector_size = 64;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector128"))
+		vector_size = 128;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector256"))
+		vector_size = 256;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector512"))
+		vector_size = 512;
+	else if (!strcmp (m_class_get_name (cmethod->klass), "Vector"))
+		vector_size = 0;
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:1354]: This is kind of mis-leading.

Comment (vargaz) [reply src/mono/mono/mini/simd-intrinsics.c:1354]: Vector is actually a different class than Vector128/Vector256/ etc., it has different methods, so ideally, it should be handled in a separate function.

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:1354]: Vector and Vector64/128/256 are now having the same sets of API's. There is no need to have a separate. Vector is an older class, comparing to Vector64/128/256.

**Review by fanyang-mono — APPROVED** [review]
Thanks for implementing this intrinsics.

**Pull request merged**